### PR TITLE
Eoracle mainnet integrated

### PR DESCRIPTION
**Pull request by RonTuretzky**
Creating a version compatible with local deploy

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,7 +2,7 @@
 cache/
 out/
 node_modules/
-
+script/output/5/eoracle_middleware_contracts_deployment_data.json
```
Comment (yonikesel1) [inline .gitignore]: Usually we want to upload the deployment data.
If we don't want to add it in this branch let's just not add the file and remove the gitignore.

File: script/DeployEOMiddlewareContracts.s.sol [hunk 2]
```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "forge-std/Script.sol";
+import "forge-std/StdJson.sol";
+import "forge-std/console.sol";
```
Comment (yonikesel1) [inline script/DeployEOMiddlewareContracts.s.sol:6]: remove console

File: script/DeployEOMiddlewareContracts.s.sol [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    IEOBLSApkRegistry,
+    IEOIndexRegistry,
+    IEOStakeRegistry,
+    IServiceManager
+} from "../src/EORegistryCoordinator.sol";
+import {EOBLSApkRegistry} from "../src/EOBLSApkRegistry.sol";
+import {EOIndexRegistry} from "../src/EOIndexRegistry.sol";
+import {EOStakeRegistry} from "../src/EOStakeRegistry.sol";
+//import {EOServiceManager} from "../src/EOServiceManager.sol";
+// Remove mock when EOServiceManager is implemented
```
Comment (yonikesel1) [inline script/DeployEOMiddlewareContracts.s.sol]: We should use EOServiceManager and not ServiceManagerMock

File: src/EORegistryCoordinator.sol [hunk 4]
```diff
@@ -474,28 +480,33 @@ contract EORegistryCoordinator is
...
 
-        // Register the operator with the BLSApkRegistry, StakeRegistry, and IndexRegistry
+        // Register the operator with the EOBLSApkRegistry, EOStakeRegistry, and EOIndexRegistry
         blsApkRegistry.registerOperator(operator, quorumNumbers);
         (results.operatorStakes, results.totalStakes) = 
             stakeRegistry.registerOperator(operator, operatorId, quorumNumbers);
         results.numOperatorsPerQuorum = indexRegistry.registerOperator(operatorId, quorumNumbers);
         if (address(EOChainManager) != address(0)){
-            EOChainManager.registerValidator(operator, results.operatorStakes);
+            if ( chainValidatorSignature.X != 0 && chainValidatorSignature.Y != 0 && chainValidatorG2Signature.X[0] != 0 && chainValidatorG2Signature.X[1] != 0 && chainValidatorG2Signature.Y[0] != 0 && chainValidatorG2Signature.Y[1] != 0){
```
Comment (ArielLaub) [inline src/EORegistryCoordinator.sol:491]: add comment
// check if is a valid chain validator signature. if not register as a data validator

File: src/EORegistryCoordinator.sol [hunk 5]
```diff
@@ -474,28 +480,33 @@ contract EORegistryCoordinator is
...
+                EOChainManager.registerValidator(operator, results.operatorStakes, [chainValidatorSignature.X ,chainValidatorSignature.Y],[chainValidatorG2Signature.X[0],chainValidatorG2Signature.X[1],chainValidatorG2Signature.Y[0],chainValidatorG2Signature.Y[1]]);
```
Comment (ArielLaub) [inline src/EORegistryCoordinator.sol]: add comment
// chain validator expects the sig and pubkey spread as arrays.

File: src/EORegistryCoordinator.sol [hunk 6]
```diff
@@ -56,9 +56,9 @@ contract EORegistryCoordinator is
     IEOChainManager public EOChainManager;
```
Comment (ArielLaub) [inline src/EORegistryCoordinator.sol]: storage vars should not start with a capital. I would not even add the prefix EO for var names just for types.
recommend renaming to chainManager instead.

Comment (ArielLaub) [reply src/EORegistryCoordinator.sol]: BTW slither also throws a warning about this.

File: script/utils/Utils.s.sol [hunk 7]
```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "../../src/interfaces/IEORegistryCoordinator.sol";
```
Comment (yonikesel1) [inline script/utils/Utils.s.sol]: remove

File: script/utils/Utils.s.sol [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import "eigenlayer-contracts/src/contracts/strategies/StrategyBase.sol";
```
Comment (yonikesel1) [inline script/utils/Utils.s.sol]: remove

File: src/interfaces/IEOBLSApkRegistry.sol [hunk 9]
```diff
@@ -29,6 +29,7 @@ interface IBLSApkRegistry is IRegistry {
      */     
     struct PubkeyRegistrationParams {
         BN254.G1Point pubkeyRegistrationSignature;
+        BN254.G1Point chainValidatorSignature;
```
Comment (yonikesel1) [inline src/interfaces/IEOBLSApkRegistry.sol:32]: is chainValidatorSignature a BLS signature?
Should we call it "BLSpubkeySignature" or something like that?

File: test/integration/tests/DeployEOMiddlewareContracts.t.sol [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+import {DeployEOMiddlewareContracts} from "../../../script/DeployEOMiddlewareContracts.s.sol";
+
+// Middleware contracts
+import {EORegistryCoordinator, IEORegistryCoordinator, IEOBLSApkRegistry, IEOIndexRegistry, IEOStakeRegistry, IServiceManager} from "../../../src/EORegistryCoordinator.sol";
+import {EOBLSApkRegistry} from "../../../src/EOBLSApkRegistry.sol";
+import {EOIndexRegistry} from "../../../src/EOIndexRegistry.sol";
+import {EOStakeRegistry} from "../../../src/EOStakeRegistry.sol";
+//import {EOServiceManager} from "../src/EOServiceManager.sol";
+// Remove mock when EOServiceManager is implemented
```
Comment (yonikesel1) [inline test/integration/tests/DeployEOMiddlewareContracts.t.sol]: Update to EOServiceManager

File: .gitignore (hunk 1 above)
Comment (yonikesel1) [reply .gitignore]: removed

File: script/DeployEOMiddlewareContracts.s.sol (hunk 2 above)
Comment (yonikesel1) [reply script/DeployEOMiddlewareContracts.s.sol:6]: removed

File: src/EORegistryCoordinator.sol (hunk 4 above)
Comment (yonikesel1) [reply src/EORegistryCoordinator.sol:491]: added

File: src/EORegistryCoordinator.sol (hunk 5 above)
Comment (yonikesel1) [reply src/EORegistryCoordinator.sol]: added

File: src/EORegistryCoordinator.sol (hunk 6 above)
Comment (yonikesel1) [reply src/EORegistryCoordinator.sol]: changed

File: script/utils/Utils.s.sol (hunk 7 above)
Comment (yonikesel1) [reply script/utils/Utils.s.sol]: removed

File: script/utils/Utils.s.sol (hunk 8 above)
Comment (yonikesel1) [reply script/utils/Utils.s.sol]: removed

File: test/integration/tests/DeployEOMiddlewareContracts.t.sol (hunk 10 above)
Comment (yonikesel1) [reply test/integration/tests/DeployEOMiddlewareContracts.t.sol]: updated

File: script/DeployEOMiddlewareContracts.s.sol (hunk 3 above)
Comment (yonikesel1) [reply script/DeployEOMiddlewareContracts.s.sol]: created new contract EOServiceManager

**Comment by yonikesel1** [comment]
Review fixes are fixed in - https://github.com/Eoracle/eoracle-middleware/pull/5

File: src/interfaces/IEOBLSApkRegistry.sol (hunk 9 above)
Comment (yonikesel1) [reply src/interfaces/IEOBLSApkRegistry.sol:32]: its ok

**Review by uri-lightblocks — APPROVED** [review]

**Pull request merged**